runtime/service_integ_test.go

```diff
@@ -533,3 +534,80 @@ vdf  254:80   0  512B  0 disk`
 			"context cancelled while waiting for container %s to exit, err: %v", containerName, ctx.Err())
 	}
 }
+
+func startAndWaitTask(ctx context.Context, t *testing.T, c containerd.Container) string {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	task, err := c.NewTask(ctx, cio.NewCreator(cio.WithStreams(nil, &stdout, &stderr)))
+	require.NoError(t, err, "failed to create task for container %s", c.ID())
+
+	exitCh, err := task.Wait(ctx)
+	require.NoError(t, err, "failed to wait on task for container %s", c.ID())
+
+	err = task.Start(ctx)
+	require.NoError(t, err, "failed to start task for container %s", c.ID())
+	defer task.Delete(ctx)
+
+	select {
+	case exitStatus := <-exitCh:
+		require.NoError(t, exitStatus.Error(), "failed to retrieve exitStatus")
+		require.Equal(t, uint32(0), exitStatus.ExitCode())
+		require.Equal(t, "", stderr.String())
+	case <-ctx.Done():
+		require.Fail(t, "context cancelled",
+			"context cancelled while waiting for container %s to exit, err: %v", c.ID(), ctx.Err())
+	}
+
+	return stdout.String()
+}
+
+func TestCreateContainerWithSameName_Isolated(t *testing.T) {
+	internal.RequiresIsolation(t)
+
+	ctx := namespaces.WithNamespace(context.Background(), "default")
+
+	client, err := containerd.New(containerdSockPath, containerd.WithDefaultRuntime(firecrackerRuntime))
+	require.NoError(t, err, "unable to create client to containerd service at %s, is containerd running?", containerdSockPath)
+	defer client.Close()
+
+	image, err := client.Pull(ctx, debianDockerImage, containerd.WithPullUnpack, containerd.WithPullSnapshotter(naiveSnapshotterName))
+	require.NoError(t, err, "failed to pull image %s, is the the %s snapshotter running?", debianDockerImage, naiveSnapshotterName)
+
+	containerName := fmt.Sprintf("%s-%d", t.Name(), time.Now().UnixNano())
+	snapshotName := fmt.Sprintf("%s-snapshot", containerName)
+
+	containerPath := fmt.Sprintf("/run/containerd/io.containerd.runtime.v2.task/default/%s", containerName)
```

Comment:
It's good to check this too, but what about the symlinks I mentioned last revision? It actually looks like we aren't deleting those at all right now AFAIK: https://github.com/firecracker-microvm/firecracker-containerd/blob/master/runtime/service.go#L774

I can see why that works out for the test case as is where you re-use a container name, but I think we should address removing those symlinks and have this test verify they are gone too. Otherwise, the shim's directory can fill up with symlinks for containers that have already been deleted. Let me know if I'm misunderstanding or missing something though.

Reply:
The shim itself is removing things from /var/run/firecracker-containerd/$namespace/$vm_id at `Shutdown()`.

https://github.com/firecracker-microvm/firecracker-containerd/blob/d9ebc8b24cac928fef30f302dae44f07489a3dbf/runtime/service.go#L999

When a VM only hosts a single task, what we have seems sufficient. When a VM hosts multiple tasks (in other words, the VM is persistent), we need to remove things at the end of the task, which we are not doing right now.

Probably I should have a different test for the persistent VM case.

Reply:
Ahhh right, that all adds up then. Yeah I agree having a test case for each codepath would be good

Reply:
Added /var/run check on https://github.com/firecracker-microvm/firecracker-containerd/pull/225/commits/31baf7a8fef63603023185d514d5da014420ce6c